SpokePoolUpgrades.md

```diff
@@ -0,0 +1,36 @@
+# An Overview of how SpokePools would be upgraded
+
+## Before executing upgrade
+
+To be safe, make sure there is no pending bundle in the HubPool. To be extra safe, disable every single deposit route
+and only re-enable them after executing the upgrade. This is probably over-cautious when upgrading via proxy but
+definitely should be done when upgrading without a proxy.
+
+## When upgrading via proxy:
+
+1. Deploy new SpokePool implementation contract on L2.
+2. Call `upgradeTo` on existing SpokePool proxy with new implementation address. The existing state should carry over
+   and the system should just work.
+
+## When upgrading without a proxy:
+
+If SpokePool is not an upgradeable proxy, then upgrading is more involved.
+
+1. Deploy new SpokePool contract.
```

Comment:
Should there be anything in here about dealing with assets that are in the spoke pool? Should they be transferred to the new spoke pool? Or back to the hub? You could send and execute a series of "relayer refunds" to transfer assets to the new spoke pool.